### ISSUE TITLE
Cannot use a copy of Ventoy stick; cannot re-create; I'm confused

### ISSUE THREAD

**Issue by kogorman** [ventoy/Ventoy#886]
I have 6 computers of various kinds.  I liked the design of Ventoy because I thought I could have one stick and it would boot on all of them, and contain all the ISOs that I use.  So far I'm mostly baffled.

I kept at it because I quickly got a stck (Patriot Rage 128GB USB 3.0) with Ventoy and it booted on all of my machines, with all the ISO's I wanted.  Yay, right?   It was 1.0.42, and GPT without Secure enabled.

For reasons I'll get to (less distressing that the main issue) I wanted to make some more.  Some to travel with my laptop, one for the garage where most of the computers are, etc.  Not so good.

Problem #1
I made a bunch of attempts that failed to one degree or another.  Eventually, I made 4 sticks on identical hardware: 16GB USB 2.0 sticks.  Each had a different combination of MBR/GPT  and Secure enabled/disabled.  None would boot on my machine with the Gigabyte motherboard, in legacy mode (booo).  All would boot on my System76 laptop, in EFI mode (important when I'm booting an ISO to do an install).  None would boot on my two Dell Optiplex 990's when I asked for EFI boot.  (The gigabyte mobo has "Hybrid EFI" so it figures that out for itself.) but they would boot in legacy mode.  They are all Ventoy 1.0.42, created on Linux with Ventoy2Disk.sh.  All have ISOs copied from the one good stick.  ISSUE: how can it be the sticks are behaving differently?

Problem #2
So I tried making a copy of the good stick.  I tried some software I had created and the result failed.  So I just used dd to copy the stick byte-for-byte onto a Mushkin USB twice as big.  It failed quicky with an error message about an unknown file system.  So I used gparted to fix the secondary GPT to cover the whole drive.  The symptoms are not always the same, but it still refuses to boot on my Gigabyte system.  It boots as well as the small sticks did on the Dells, And boots UEFI on my System76.  ISSUE: Why not the Dells, and why does an exact copy behave differently?

I'm at a loss for how to explore this further, but will take suggestions.

Feature request:
I'd like to be able to create a stick that WILL NOT boot UEFI, for systems that detect and prefer UEFI.  That's for times I want to use the ISO to create a non-EFI install.

**Comment by kogorman** [comment]
The biggest issue I have is simply this:
I have one working jump drive that works in all machines, but I have been unable to make another, or make a working copy.

**Comment by kogorman** [comment]
The next most important is finding a way to create a UEFI-only stick.  My main system boots xubuntu install CDs as UEFI, but when it's uinder Ventoy, the CD runs as legacy.  I want to use that system to create UEFI boot disks for other machines.

**Comment by steve6375** [comment]
There are a great many fake USB drives about. Test them with FakeFlashTest.exe. Because Ventoy puts the boot files at the end of the disk, fake USB drives won't work.
https://rmprepusb.com/tutorials/007-all-about-fake-sd-cards-and-usb-flash-drives/
If they pass the test, I would start by simply using the Ventoy sh script to make Ventoy USB drives but do NOT copy any ISOs or other files over. Then see if it both Legacy and MBR-boots.
To prevent a system from always EFI booting, delete (or rename) the \EFI folder on the second partition.

**Comment by kogorman** [comment]
Yeah, you were right.  I was using fake drives because they were the only ones with multiples the same size.  I'm now using a more reliable kind.

I'm okay with EFI booting.  Indeed I want to insist on it.  Can I delete or rename something to prevent legacy booting?

**Comment by steve6375** [comment]
There is usually no need to prevent legacy booting. If a system tries to legacy boot and no valid boot code is present, you would get a BIOS error message. So with no valid boot code you get an BIOS error message if the BIOS tries to legacy-boot, with valid boot code it would legacy boot to Ventoy.

**Comment by kogorman** [comment]
That does not work on all of my machines, specifically a Gigabyte mobo with what it calls "Hybrid EFI Technology".   It picks UEFI on my CDs, but the ISO inside a Ventoy that allows both gets booted in legacy mode.  That means I cannot use an install ISO under Ventoy to install an EFI system.

I'm not sure what's going wrong, but I'd like to try disabling legacy on the Ventoy stick.

**Comment by kogorman** [comment]
The fix for an installed OS is here but I cannot see how to apply it to Ventoy: https://askubuntu.com/questions/1279602
